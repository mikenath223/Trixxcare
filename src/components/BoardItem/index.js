import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const BoardItem = ({
  doc: {
    image, name, id, details,
  }, style,
}) => {
  const preventDrag = e => e.preventDefault();

  return (
    <div className={style.docWrap} onDragStart={preventDrag}>
      <Link to={`doctors/${name}-${id}`}>
        <img className={style.image} src={image} alt="doctor" />
        <p className={style.docName}>{name}</p>
        <p className={style.docDet}>{details.split('|')[0]}</p>
        <p>
          <img src="https://img.icons8.com/carbon-copy/30/000000/facebook-circled.png" alt="" />
          <img src="https://img.icons8.com/wired/30/000000/twitter-circled.png" alt="" />
          <img src="https://img.icons8.com/carbon-copy/30/000000/filled-message.png" alt="" />
        </p>
      </Link>
    </div>
  );
};

BoardItem.propTypes = {
  doc: PropTypes.shape({
    name: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    details: PropTypes.string.isRequired,
  }).isRequired,
  style: PropTypes.shape({
    docDet: PropTypes.string,
    docName: PropTypes.string,
    image: PropTypes.string,
    docWrap: PropTypes.string,
  }).isRequired,
};

export default BoardItem;
