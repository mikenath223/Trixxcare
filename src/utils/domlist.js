const showRegisterForm = () => {
  const form = document.querySelector('.reg-form');
  const regModal = document.querySelector('.reg-modal');

  document.querySelector('.close-reg').onclick = () => {
    regModal.style.visibility = 'hidden';
    form.style.transform = 'translateY(-100%)';
  };
  window.onclick = e => {
    const modals = document.querySelectorAll('.modal');
    const forms = document.querySelectorAll('.form');
    modals.forEach(el => {
      if (e.target === el) {
        const elem = el;
        elem.style.visibility = 'hidden';
        forms.forEach(fr => {
          const form = fr;
          form.style.transform = 'translateY(-100%)';
          form.style.opacity = 0;
          return '';
        });
      }
    });
  };
};

const showSigninForm = () => {
  const form = document.querySelector('.signin-form');
  const signModal = document.querySelector('.signin-modal');

  document.querySelector('.close-sign').onclick = () => {
    signModal.style.visibility = 'hidden';
    form.style.transform = 'translateY(-100%)';
    form.style.opacity = 0;
  };
};

const showRegister = () => {
  const form = document.querySelector('.reg-form');
  const signModal = document.querySelector('.reg-modal');
  showRegisterForm();
  signModal.style.visibility = 'visible';
  form.style.transform = 'translateY(30px)';
  form.style.opacity = 1;
};

const showSignin = () => {
  const form = document.querySelector('.signin-form');
  const regModal = document.querySelector('.signin-modal');
  showRegisterForm();
  showSigninForm();
  regModal.style.visibility = 'visible';
  form.style.transform = 'translateY(30px)';
  form.style.opacity = 1;
};

const slideHeight = () => {
  document.querySelector('.mainContainer').style = 'position: fixed; height: 100%; top: 0; left: 0; background-color: #FFB400;';
  document.querySelector('.slideContainer').style = 'height: 100%';
  document.querySelector('.slidesHolder').style = 'height: 100%';
  document.querySelectorAll('.but').forEach(el => {
    const elem = el;
    elem.type = 'submit';
    return '';
  });
};

const setText = () => {
  document.querySelector('.alice-carousel__prev-btn-item').innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
        width="24" height="24"
        viewBox="0 0 172 172"
        style={{ fill: "#000000;" }}><g fill="none" fill-rule="nonzero" stroke="none" stroke-width="1" strokeLinecap="butt" strokeLinejoin="miter" strokeMiterlimit="10" strokeDasharray="" strokeDashoffset="0" font-family="none" font-weight="none" font-size="none" text-anchor="none" style={{ mixBlendMode: "normal" }}><path d="M0,172v-172h172v172z" fill="none"></path><g id="original-icon" fill="#ffffff"><path d="M86,39.1085l55.255,89.8915h-110.51l55.255,-89.8915M86,21.5c-3.54033,0 -7.08783,1.77017 -9.25933,5.30333l-60.62283,98.61333c-4.74433,7.72567 0.52317,17.91667 9.2665,17.91667h121.2385c8.74333,0 14.01083,-10.191 9.25933,-17.91667l-60.62283,-98.61333c-2.1715,-3.53317 -5.719,-5.30333 -9.25933,-5.30333z"></path></g></g></svg>`;
  document.querySelector('.alice-carousel__next-btn-item').innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
        width="24" height="24"
        viewBox="0 0 172 172"
        style={{ fill: "#000000;" }}><g fill="none" fill-rule="nonzero" stroke="none" stroke-width="1" strokeLinecap="butt" strokeLinejoin="miter" strokeMiterlimit="10" strokeDasharray="" strokeDashoffset="0" font-family="none" font-weight="none" font-size="none" text-anchor="none" style={{ mixBlendMode: "normal" }}><path d="M0,172v-172h172v172z" fill="none"></path><g id="original-icon" fill="#ffffff"><path d="M86,39.1085l55.255,89.8915h-110.51l55.255,-89.8915M86,21.5c-3.54033,0 -7.08783,1.77017 -9.25933,5.30333l-60.62283,98.61333c-4.74433,7.72567 0.52317,17.91667 9.2665,17.91667h121.2385c8.74333,0 14.01083,-10.191 9.25933,-17.91667l-60.62283,-98.61333c-2.1715,-3.53317 -5.719,-5.30333 -9.25933,-5.30333z"></path></g></g></svg>`;
};

const handleOpenMenu = () => {
  const sideBar = document.querySelector('.sideBar');
  sideBar.style.display = 'flex';
};

const handleCloseMenu = () => {
  document.querySelector('.sideBar').style.display = 'none';
};


const resizer = () => {
  window.addEventListener('resize', () => {
    if (document.querySelector('.sideBar')) {
      if (window.innerWidth >= 710) {
        document.querySelector('.sideBar').style.display = 'flex';
      } else {
        document.querySelector('.sideBar').style.display = 'none';
      }
    }
  });
};

export {
  showRegister, showSignin, slideHeight, setText, handleOpenMenu, handleCloseMenu, resizer,
};
