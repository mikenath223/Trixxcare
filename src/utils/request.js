/* 
/ Util function to get user token
*/
export const getToken = (username, password, trigger,
  setAuth, setAlert, setLoad) => {
  fetch('https://trixxcare.herokuapp.com/api/v1/user_token', params({
    auth: {
      username, password,
    },
  }))
    .then(res => {
      if (res.status === 201) {
        res.json().then(token =>
          saveToken(token, username, setAuth, trigger));
      } else {
        setAlert(prevState => ({
          ...prevState,
          message: 'Username or password is incorrect!',
          load: true,
        }));
      }
      setLoad(false);
    });
};

const params = (paramsBody) => ({
  method: 'POST',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
  },
  body: JSON.stringify(paramsBody),
})
const saveToken = (token, username, setAuth, trigger) => {
  localStorage.setItem('tok', JSON.stringify(token.jwt));
  setAuth({ user: username, isLogged: true });
  hideForms()
  trigger({ show: false });
}
const hideForms = () => {
  const modals = document.querySelectorAll('.modal');
  const forms = document.querySelectorAll('.form');
  forms.forEach(el => {
    const elem = el;
    elem.classList.remove('translate');
    return '';
  });
  modals.forEach(el => {
    const elem = el;
    elem.style.visibility = 'hidden';
    return '';
  });
}


/* 
/ Util function to register new user
*/
export const registerUser = (username, password,
  trigger, setAlert, setLoad, setSigninCred, setAuth) => {
  fetch('https://trixxcare.herokuapp.com/api/v1/users', params({
    username, password,
  }))
    .then(res => {
      if (res.status !== 204) {
        res.json().then(rep => {
          setAlert(prevState => ({
            ...prevState,
            message: rep.message,
            load: true,
            type: 'error',
          }));
          setLoad(false);
        });
      } else {
        getToken(username, password, trigger,
          setAuth, setAlert, setLoad);
        setSigninCred({ username: '', password: '' })
      }
    })
    .catch(() => {
      setAlert(prevState => ({
        ...prevState,
        message: 'Unexpected error!',
        load: true,
        type: 'error',
      }));
      setLoad(false);
    });
}

/* 
/ Util function to get current user
*/
export const getCurrentUser = (setAuth, setAlert) => {
  fetch('https://trixxcare.herokuapp.com/api/v1/currentuser', {
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${JSON.parse(localStorage.tok)}`,
    },
  }).then(res => res.json())
    .then(usr => {
      setAuth({ isLogged: true, user: usr.user });
    })
    .catch(() => {
      setAlert(prevState => ({
        ...prevState,
        message: 'Unexpected error!',
        load: true,
      }));
    });
}

/* 
/ Util function to get caregivers
*/
export const getCaregivers = (setDocs, setIsLoaded, setErr, setText) => {
  fetch('https://trixxcare.herokuapp.com/api/v1/doctors', {
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
  }).then(res => res.json())
    .then(res => {
      setDocs(res);
      setIsLoaded(true);
      setText();
    })
    .catch(() => {
      setIsLoaded(true);
      setErr(true);
    });
}

/* 
/ Util function to get all current user appointments
*/
export const getAppointments = (setAppoints, setIsLoaded, setErr) => {
  fetch('https://trixxcare.herokuapp.com/api/v1/appointments', {
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${JSON.parse(localStorage.tok)}`,
    },
  }).then(res => res.json())
    .then(res => {
      setAppoints(res);
      setIsLoaded(true);
    })
    .catch(() => {
      setIsLoaded(true);
      setErr(true);
    });
}

/* 
/ Util function to set appointments
*/
export const setAppointment = (token, date, id, ret, setAlert,
  alert, setShowpay, showPay, setDate) => {
  fetch('https://trixxcare.herokuapp.com/api/v1/appointments', {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${JSON.parse(token)}`,
    },
    body: JSON.stringify({
      date,
      doctor_id: id,
      location: ret.location,
      doctor_name: ret.name,
    }),
  }).then(res => {
    if (res.status !== 204) {
      res.json().then(rep => {
        setAlert({
          ...alert,
          message: rep.message,
          load: true,
          type: 'error',
        });
      });
    }
    else {
      setAlert({
        ...alert,
        message: 'Appointment created!',
        load: true,
        type: 'success',
      });
      setShowpay({
        ...showPay,
        payBtn: true,
      });
    }
    setDate('');
  }).catch(e => {
    setAlert({
      ...alert,
      message: e.message,
      load: true,
      type: 'error',
    });
  });
}

/* 
/ Util function to delete appointments
*/
export const deleteAppoints = (appId, setAlert, delAppoints) => {
  fetch(`https://trixxcare.herokuapp.com/api/v1/appointments/${appId}`, {
    method: 'DELETE',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${JSON.parse(localStorage.tok)}`,
    },
  }).then(res => {
    if (res.status !== 204) {
      res.json().then(rep => {
        setAlert(prevState => ({
          ...prevState,
          message: rep.message,
          load: true,
          type: 'error',
        }));
      });
    }
    else {
      delAppoints(appId);
      setAlert(prevState => ({
        ...prevState,
        message: 'Appointment deleted!',
        load: true,
        type: 'success',
      }));
    }
  }).catch(e => {
    setAlert(prevState => ({
      ...prevState,
      message: e.message,
      load: true,
      type: 'error',
    }));
  });
}

/* 
/ Util function to get single caregiver
*/
export const getCaregiver = (id, setRet, setIsLoaded, setErr) => {
  fetch(`https://trixxcare.herokuapp.com/api/v1/doctors/${id}`, {
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      Authorization: `Bearer ${JSON.parse(localStorage.tok)}`,
    },
  }).then(res => res.json())
    .then(res => {
      setRet(res);
      setIsLoaded(true);
    })
    .catch(() => {
      setIsLoaded(true);
      setErr(true);
    });
}