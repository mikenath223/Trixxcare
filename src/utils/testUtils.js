import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import rootReducer from 'store/index';

const store = createStore(rootReducer);
export const withReduxRendering = component => (
  render(
    <Provider store={store}>
      {component}
    </Provider>,
  ));

export const withRouter = component => (
  render(
    <MemoryRouter>
      {component}
    </MemoryRouter>,
  )
);

const withReduxRouter = component => (
  withReduxRendering(
    <MemoryRouter>
      {component}
    </MemoryRouter>,
  ));

export default withReduxRouter;
