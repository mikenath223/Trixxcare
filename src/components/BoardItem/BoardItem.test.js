import React from 'react';
import { withRouter } from 'utils/testUtils';
import { cleanup } from '@testing-library/react';
import BoardItem from './index';

describe('BoardItem', function () {
  afterEach(cleanup);

  beforeEach(() => {
    this.props = {
      doc: {
        image: 'image',
        id: 1,
        details: 'Orthopedic',
        name: 'John Doe',
      },
      style: {},
    };
  });

  it('should match snapshot', () => {
    const { container } = withRouter(<BoardItem {...this.props} />);

    expect(container).toMatchSnapshot();
  });

  it('should display details', () => {
    const { getByText } = withRouter(<BoardItem {...this.props} />);

    expect(getByText(/John Doe/)).toBeInTheDocument();
  });
});
