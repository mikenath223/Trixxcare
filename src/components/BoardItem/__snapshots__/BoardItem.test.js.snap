// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardItem should match snapshot 1`] = `
<div>
  <div>
    <a
      href="/doctors/John Doe-1"
    >
      <img
        alt="doctor"
        src="image"
      />
      <p>
        John Doe
      </p>
      <p>
        Orthopedic
      </p>
      <p>
        <img
          alt=""
          src="https://img.icons8.com/carbon-copy/30/000000/facebook-circled.png"
        />
        <img
          alt=""
          src="https://img.icons8.com/wired/30/000000/twitter-circled.png"
        />
        <img
          alt=""
          src="https://img.icons8.com/carbon-copy/30/000000/filled-message.png"
        />
      </p>
    </a>
  </div>
</div>
`;
